Dbm fastmod ec

dbmcclain opened the pull request.

easye commented on src/Cosi-BLS/range-proofs.lisp:
```diff
@@ -693,421 +695,578 @@ nbits pts will become right vectors."
 ;; Constants obtained from:
 
-       (with-ed-curve :curve-1174
+       (with-ed-curve :curve1174
```
I kinda like the style of using `#\-` `for-separating-words` better, but not necessary to change this now.

Your usage here is certainly consistent with how the C code names things, which doesn't have the ability to use `#\-" as such a separator.

dbmcclain replied:
My naming conventions follow the academic literature (or try to), so Curve1174 is its name in print, as is Ed3363, where I do separate as curve-Ed3363 in order to keep it clear in the code that it is an ECC curve.

easye replied:
Fine then.

easye commented on src/Crypto/core-crypto.asd:
```diff
@@ -37,16 +37,16 @@ THE SOFTWARE.
-                #+:COM.RAL (:file "crypto-le")
-                (:file "kdf")
-                (:file "gf-571")
+                #+(AND :MACOSX :COM.RAL) (:file "crypto-le")
```
It would be nice to remove or start deprecating the use of the `:com.ral` for `cl:*features*`.  

The use of `:com.ral` needs to be explained to every new developer, and isn't clear what it means.  

Since you always have the `:com.ral` feature present when you develop, I think it quite likely that there will eventually be some kind of bug that bites us because you are the only one that uses that code path.

dbmcclain replied:
yes started thinking about that last night as I went weeding... Perhaps time to put a clear separation between my other uses of the crypto libs

easye replied:
Again, not necessary for this pull request, but should be done sooner rather than later.

easye commented on src/Crypto/core-crypto.asd:
```diff
@@ -55,5 +55,48 @@ THE SOFTWARE.
+                 "crypto-pairings/libraries"
                  ))
 
+(defsystem "core-crypto/libraries"
```
Probably better to start a new hierarchy for the various external libraries that `core-crypto` needs.

Perhaps:

 "core-crypto/libraries" --> "core-crypto/lib/curve1174"
"crypto-pairings/libraries" --> "crypto-pairings/lib/pbc"

dbmcclain replied:
good idea

easye commented on src/Crypto/crypto-pairings.asd:
```diff
@@ -26,37 +26,14 @@ THE SOFTWARE.
   :version     "1.1.0"
   :author      "D.McClain <dbm@refined-audiometrics.com>"
-  :in-order-to ((test-op (test-op "crypto-pairings-test")))
+  :in-order-to ((test-op (test-op "crypto-pairings-tests")))
```
Please use the singular for for packages and ASDF systems to be consistent:

"crypto-pairings-tests" --> ""crypto-pairings-test"

dbmcclain replied:
yes, I thought I did that yesterday. I like your consistent convention. Looks like I need to go back and double check what I thought I did already.

dbmcclain replied:
fixed

easye commented on src/Crypto/core-crypto.asd:
```diff
@@ -22,13 +22,13 @@ OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
   :description "core-crypto: core cryptography functions"
   :version     "1.0"
   :author      "D.McClain <dbm@refined-audiometrics.com>"
+  :in-order-to ((test-op (test-op "core-crypto-tests")))
```
```
(asdf:test-system :core-crypto)
```
fails because this line should be 

```
:in-order-to ((test-op (test-op "core-crypto-test")))
```

dbmcclain replied:
fixed

dbmcclain commented:
Thanks for taking a look at this stuff, Mark.

easye commented on src/Crypto/crypto-pairings-test.asd:
```diff
@@ -4,7 +4,7 @@
                crypto-pairings)
   :perform  (test-op (o s)
-                     (symbol-call :lisp-unit :run-tests
+                     (symbol-call :lisp-unit :run-test
```
@dbmcclain Actually, that needs to be `:run-tests` not `:run-test`, as thats the API of `lisp-unit` which actually runs the tests.

dbmcclain commented:
okay we are significantly cleaned up and ready to merge into dev at your convenience

Trying to see what happened from the diff log will be a chore. What I did was to move all unused Crypto sources into a separate Emotiq-Unused folder. That is not being tracked, and I stopped the tracking on all those individual files.

If we ever need one of them again, I'll move it up into the Crypto folder again and restart tracking on it.

I also moved all our PBC-Interface, and ECC Curves, C-code into a Crypto-Libraries folder. They remain in separate folders because of significant differences in the makefile g++ command flags.

I did the same cleanup of old files for the Randhound folder.

And I fixed a few problems, sped up the Curve1174 modular inverse, and added test lines to the etc/all-tests file
